File: clean/proc/Info.cs
using System;

namespace clean.proc
{
    internal class Info
    {
        int version;
        byte[] title = new byte[1024];
        byte[] author = new byte[1024];
        Time createTime;
        Time updateTime;

        void SetDefault()
        {
            version = 0;
            title[0] = 0x81;
            title[1] = 0x00;
            author[0] = 0x81;
            author[1] = 0x00;
            createTime.SetDefault();
            updateTime.SetDefault();
        }
    }
}

File: src/info.c
#include "info.h"

#include <stdio.h>
#include <string.h>
#include "proc.h"

void init_info(info_t *info){
    init_proc(&(info->tag));
    info->version = 1;
    strcpy(info->title, "default title");
    strcpy(info->author, "default author");
    return;
}

void print_info(const info_t *info){
    printf("version: %d\n", info->version);
    printf("title: %s\n", info->title);
    printf("author: %s\n", info->author);
    return;
}
